sceptre/connection_manager.py

```diff
@@ -157,7 +164,8 @@ def _get_session(self, profile, region, iam_role):
                     session_name = f'{iam_role.split("/")[-1][:56]}-session'
                     sts_response = sts_client.assume_role(
                         RoleArn=iam_role,
-                        RoleSessionName=session_name
+                        RoleSessionName=session_name,
+                        DurationSeconds=self.iam_role_session_duration
                     )
```

Comment:
Based upon how it's instantiated, `self.iam_role_session_duration` is going to be `None` sometimes, which you've already said IAM rejects. Therefore, I think you should do something like this:
```python
assume_role_kwargs = {
    'RoleArn': iam_role,
    'RoleSessionName': session_name,
}
if self.iam_role_session_duration:
    assume_role_kwargs['DurationSeconds'] = self.iam_role_session_duration
sts_response = sts_client.assume_role(**assume_role_kwargs)
```